### ISSUE TITLE
umi开启pwa缓存不会更新

### ISSUE THREAD

**Issue by xiayu1986** [umijs/umi#4339]
umi版本：2.13.9
问题描述：开启PWA之后，发现缓存不会刷新，文件已经重新生成了，但是刷新后访问的仍然是上一个版本
umi配置：pwa: {
        workboxPluginMode: 'GenerateSW',
        workboxOptions: {
          importWorkboxFrom: 'local',
        },
        manifestOptions: {
          srcPath: 'public/manifest.json'
        }
      }

**Comment by QianDingwei** [comment]
大佬，想请教你一下，开启pwa是只需要config中配置嘛，我的问题是配置完了，但是好像servicework并没有注册，在serviceworker中console.log都没有输入，这是为啥，麻烦指教

**Comment by sherry007** [comment]
> 大佬，想请教你一下，开启pwa是只需要config中配置嘛，我的问题是配置完了，但是好像servicework并没有注册，在serviceworker中console.log都没有输入，这是为啥，麻烦指教

如果是在 umi3 的话，现在已经去掉 pwa 配置啦，至于现在怎么配置，可以参考下我刚刚“研究”出来的配置：https://github.com/umijs/umi/issues/4793#issuecomment-667812240

**Comment by sherry007** [comment]
> Sherry007 I am also using ANTD UMI for a new enterprise app and want to enable PWA\Offline. Thoughts on collaborating to put together article and details to share with others on how to configure PWA\Offline.

No problem, I'd love to do this together

**Comment by takeitforwardsolutions** [comment]
Hi
I got this working using Version 5 and will try with version 4. Attached raw files need to clean-up and finalize but thought I would share. I ended up doing the following with some extra touches, combining many of things you have suggested > Thank You.

1) Added workbox
2) Added workbox-webpack-plugin to config.ts injected the file
3) Created early phases of a PWA Mode and Status component and added that as a component to the right content component. To display messages, status and add a button for going online\offline\sync 
4) Created service-worker.js and service-worker-register > will register and dispatch events that can be consumed by other components for example the PWAStatus component

[config.ts.txt](https://github.com/umijs/umi/files/5226454/config.ts.txt)
[index.tsx.txt](https://github.com/umijs/umi/files/5226455/index.tsx.txt)
[service-worker.js.txt](https://github.com/umijs/umi/files/5226456/service-worker.js.txt)
[service-worker-register.ts.txt](https://github.com/umijs/umi/files/5226457/service-worker-register.ts.txt)
